internal/pkg/gateway/api.go

```diff
@@ -134,12 +134,28 @@ func (gs *Server) Endorse(ctx context.Context, request *gp.EndorseRequest) (*gp.
 	}
 	proposal, err := protoutil.UnmarshalProposal(signedProposal.ProposalBytes)
 	if err != nil {
-		return nil, status.Errorf(codes.InvalidArgument, "failed to unpack transaction proposal: %s", err)
+		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
-	channel, chaincodeID, hasTransientData, err := getChannelAndChaincodeFromSignedProposal(signedProposal)
+	header, err := protoutil.UnmarshalHeader(proposal.Header)
```

Comment:
Can't getChannelAndChaincodeFromSignedProposal() be adapted to be used by both Evaluate() and Endorse()? I liked the symmetry and re-use you had originally, not to mention keeping the Endorse() logic clean.